docs/wip/theme-shadowing.md

```diff
@@ -1,7 +1,66 @@
-## Why it is important?
+# Shadowing
+This documentation will address Shadowing in Gatsby Themes focusing on how we use it in the Store Framework Jamstack (SFJ). You can find general explanations and examples of the Shadowing concept in the [Gatsby documentation](https://www.gatsbyjs.com/docs/how-to/plugins-and-themes/shadowing/).
 
-## How to use theme shadowing on SFJ
-Explain how to overwrite components of themes
+## Importance of Shadowing
+In [Gatsby documentation](https://www.gatsbyjs.com/docs/conceptual/plugins-themes-and-starters/) we see that _"Gatsby themes allow Gatsby site functionality to be packaged as a standalone product for others (and yourself!) to easily reuse."_, it means that when you use the theme we developed, you will be able to use pre-configured functionality, data sourcing, design and everything else we add to our theme.
+
+Our theme packages already includes a site structure with a handful of pages and React components that are used to build those pages. You can use this structure to guide the development of specific pages and components of your site. In order to be able to change what we offer, you can make use of the Shadowing concept.
+
+As stated in the [Gatsby documentation](https://www.gatsbyjs.com/docs/how-to/plugins-and-themes/shadowing/): _"This feature allows users to replace a file in the src directory that is included in the webpack bundle with their own implementation."_. Therefore, you have the ability to adapt any file to suit your needs, for example: styles, to match your visual identity, and even change the functionality of a component, to meet your business rules.
+
+## Shadowing on SFJ
+
+### What can I shadow?
+In the [Gatsby documentation](https://www.gatsbyjs.com/docs/how-to/plugins-and-themes/shadowing/#any-source-file-is-shadowable) you'll learn that you can override (_shadow_) any source file, not just React components. In our example-store implementation, we shadowed this typescript [file](https://github.com/vtex/faststore/blob/master/packages/gatsby-plugin-i18n/src/i18n/en.ts) from `gatsby-plugin-i18n` to add our example-store expressions, transforming it into this [file](https://github.com/vtex-sites/storecomponents.store/blob/master/src/%40vtex/gatsby-plugin-i18n/i18n/en.ts). You can customize any file by creating a file in the same path as the theme you are using, example: `.../i18n/en.ts`, that way the Shadowing API will know that it should use your file, instead of the theme's file.
```

Comment:
```suggestion
Gatsby allows you to shadow (a.k.a., override) not just React components but also any source file.

>ℹ️ We strongly encourage you to check Gatsby's documentation and learn more about [shadowing source files.](https://www.gatsbyjs.com/docs/how-to/plugins-and-themes/shadowing/#any-source-file-is-shadowable)

Take the [Store Components](https://github.com/vtex-sites/storecomponents.store) theme as an example: to fulfill our specific needs regarding translations, we shadowed [this typescript file](https://github.com/vtex/faststore/blob/master/packages/gatsby-plugin-i18n/src/i18n/en.ts) from the Gatsby internationalization plugin (`gatsby-plugin-i18n`).

As a result, we now have [this file](https://github.com/vtex-sites/storecomponents.store/blob/master/src/%40vtex/gatsby-plugin-i18n/i18n/en.ts) on our theme, including our definitions:

```ts
export default {
  ...en,

  'shelf.title.0': 'New Offers',

  'social.share': 'Share',

  'offer.product-unavailable': 'Product Unavailable',
  'offer.units-left': '{quantity} units left!',
  'offer.installments': 'Up to {numberOfInstallments}x {value} interest-free',
  'offer.discount': ' Save {price}',

  'productDetails.reference': 'Reference',
  'product-not-found': 'Product not found',
  'error-generic': 'Error',
}
```
```